shared/src/domain-logic/opportunityFields.ts

```diff
@@ -10,16 +10,55 @@ export const OPPORTUNITY_FIELDS: (keyof ProfileData)[] = [
 
 export const REPORT_FIELDS: (keyof ProfileData)[] = ["dateOfFormation"];
 
+export const FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION: (keyof ProfileData)[] = ["municipality"];
+export const FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION: (keyof ProfileData)[] = ["homeBasedBusiness"];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
 export const FIELDS_FOR_PROFILE = [...REPORT_FIELDS, ...OPPORTUNITY_FIELDS];
-export const getFieldsForProfile = (legalStructureId?: string): (keyof ProfileData)[] => {
-  return FIELDS_FOR_PROFILE.filter((field) => {
+
+export const getFieldsForProfile = (profileData: ProfileData): (keyof ProfileData)[] => {
+  if (profileData.businessPersona === "FOREIGN" && profileData.foreignBusinessType === "NEXUS") {
+    return profileData.nexusLocationInNewJersey
+      ? lookupByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION, profileData.legalStructureId)
+      : lookupByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION, profileData.legalStructureId);
+  } else {
+    return lookupByLegalStructure(FIELDS_FOR_PROFILE, profileData.legalStructureId);
+  }
+};
+
+const lookupByLegalStructure = (
+  key: (keyof ProfileData)[],
+  legalStructureId?: string
+): (keyof ProfileData)[] => {
+  return key.filter((field) => {
     if (field === "dateOfFormation" && legalStructureId) {
       return LookupLegalStructureById(legalStructureId).elementsToDisplay.has("formationDate");
     }
     return true;
   });
 };
 
+export const isEveryFundingFieldAnswered = (profileData: ProfileData): boolean => {
+  let fields;
+  if (profileData.businessPersona === "FOREIGN" && profileData.foreignBusinessType === "NEXUS") {
+    fields = profileData.nexusLocationInNewJersey
+      ? FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION
+      : FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION;
+  } else {
+    fields = OPPORTUNITY_FIELDS;
```

Comment:
[pebble[ On the same comment thread as in `updateSidebarCards` regarding removing this helper function... 

I think another option would be to keep this function if you want (abstracting away the .every call) but use `getFieldsForProfile` inside of here instead of redeclaring the logic?

I'm really confused why we're redeclaring how to define `fields` when we have this exact same logic already written 20 lines up in `getFieldsForProfile` - am I missing something? Please let me know.

It seems like this rewritten logic is also incomplete - like, wouldn't it return the wrong set of fields for a Poppy LLC? It would be missing the `dateOfFormation` field because this logic isn't a complete duplication

Reply:
So the previous logic had date of formation not in the array when displaying the goToProfileNude bc it was only using the OpportunityFields array.  I thought we wanted to keep that logic.

Reply:
That might of been a bug as well

Reply:
done